app/models/site.rb

```diff
@@ -4,35 +4,73 @@ class Site < ActiveRecord::Base
   
   # Redis-backed properties
   include Redis::Objects
-  hash_key :proxy_pool
+  sorted_set :proxy_pool
+  lock :proxy_pool, expiration: 60.seconds
   sorted_set :proxy_successes
   sorted_set :proxy_failures
   
+  def select_proxy(older_than=-1)
```

Comment:
Add a comment describing the method, particularly what the significance of the default value of older_than.